tests/tests/test_state_scripts.py

```diff
@@ -348,71 +363,74 @@
-                "ArtifactFailure_Enter_22",
-                "ArtifactFailure_Enter_33",
-                "ArtifactFailure_Leave_44",
-                "ArtifactFailure_Leave_55",
-            ],
-        },
-    ),
+    # TODO: Move this test case out of the rest, requires full rootfs update
+    # (
+    #     "Corrupted_script_version_in_etc",
+    #     {
+    #         "FailureScript": [],
+    #         "ExpectedStatus": "failure",
+    #         "CorruptEtcScriptVersionInUpdate": True,
+    #         "ScriptOrder": [
+    #             "Idle_Enter_08_testing",
+    #             "Idle_Enter_09",
+    #             "Idle_Leave_09",
+    #             "Idle_Leave_10",
+    #             "Sync_Enter_02",
+    #             "Sync_Enter_03",
+    #             "Sync_Leave_04",
+    #             "Sync_Leave_15",
+    #             "Download_Enter_12",
+    #             "Download_Enter_13",
+    #             "Download_Leave_14",
+    #             "Download_Leave_25",
+    #             "ArtifactInstall_Enter_01",
+    #             "ArtifactInstall_Enter_02",
+    #             "ArtifactInstall_Leave_01",
+    #             "ArtifactInstall_Leave_03",
+    #             "ArtifactReboot_Enter_01",
+    #             "ArtifactReboot_Enter_11",
+    #             "ArtifactReboot_Leave_01",
+    #             "ArtifactReboot_Leave_89",
+    #             "ArtifactReboot_Leave_99",
+    #             "ArtifactCommit_Enter_01",
+    #             "ArtifactCommit_Enter_05",
+    #             "ArtifactCommit_Error_91",
+    #             "ArtifactRollback_Enter_00",
+    #             "ArtifactRollback_Enter_01",
+    #             "ArtifactRollback_Leave_00",
+    #             "ArtifactRollback_Leave_01",
+    #             "ArtifactRollbackReboot_Enter_00",
+    #             "ArtifactRollbackReboot_Enter_99",
+    #             "ArtifactRollbackReboot_Leave_01",
+    #             "ArtifactRollbackReboot_Leave_99",
+    #             "ArtifactFailure_Enter_22",
+    #             "ArtifactFailure_Enter_33",
+    #             "ArtifactFailure_Leave_44",
+    #             "ArtifactFailure_Leave_55",
+    #         ],
+    #     },
+    # ),
 ]
 
 
 REBOOT_TEST_SET = [
     (
         "simulate_powerloss_artifact_install_enter",
         {
-            "RebootScripts": ["ArtifactInstall_Enter_01",],
+            "RebootScripts": ["ArtifactInstall_Enter_02",],
             "ExpectedFinalPartition": ["OriginalPartition"],
             "ScriptOrder": [
                 "ArtifactInstall_Enter_01",
+                "ArtifactInstall_Enter_02",
                 "ArtifactInstall_Leave_01",
                 "ArtifactReboot_Enter_01",
                 "ArtifactReboot_Leave_01",
                 "ArtifactFailure_Enter_01",
                 "ArtifactFailure_Leave_89",
             ],
             "ExpectedScriptFlow": [
-                "ArtifactInstall_Enter_01",  # kill!
+                "ArtifactInstall_Enter_01",  # run one script to init log
```

Comment:
TBH I didn't get to the bottom of this one, but without initializing the `/data/test_state_scripts.log` before the reboot, the test hangs forever (until timeout, really). I think my workaround does not modify the test intention, so it should be okey.